server/services/event/getEventDetails.ts

```diff
@@ -118,21 +119,24 @@ export interface AdminEventGetResponse extends Pick<Event, typeof adminEventGetE
 
 export type EventGetResponseType<A extends boolean> = true extends A ? AdminEventGetResponse : EventGetResponse;
 
+/**
+ * @param slugOrId Event id if admin === true, slug if admin === false.
+ * @param admin Whether or not to return results for the admin view.
+ */
 export default async function getEventDetails<A extends boolean>(
```

Comment:
Why is `A` even a type parameter? :D

Reply:
This is so that we can return a different type based on whether we're getting stuff as an admin or user. Might be cleaner as a cast, but would lose typechecking.